controllers/pkg/giteaclient/giteaclient.go

```diff
@@ -36,10 +37,33 @@ type GiteaClient interface {
 	Get() *gitea.Client
 }
 
-func New(client resource.APIPatchingApplicator) GiteaClient {
-	return &gc{
-		client: client,
+var lock = &sync.Mutex{}
+
+var singleInstance *gc
+
+func GetGiteaClientInstance(ctx context.Context, client resource.APIPatchingApplicator) (GiteaClient, error) {
```

Comment:
I think just "GetClient" is more idiomatic - we already know it's a Gitea client from the package, and "Get" hints that it's retrieving rather than creating, so "Instance" seems redundant.

Reply:
Sounds good